jwst/resample/resample_step.py

```diff
@@ -114,7 +116,14 @@ def process(self, input):
             util.update_s_region_imaging(model)
             model.meta.asn.pool_name = input_models.asn_pool_name
             model.meta.asn.table_name = input_models.asn_table_name
-            model.meta.resample.pixel_scale_ratio = self.pixel_scale_ratio
+
+            # if pixel_scale exists, it will override pixel_scale_ratio.
+            # calculate the actual value of pixel_scale_ratio based on pixel_scale
+            # because source_catalog uses this value from the header.
+            if not self.pixel_scale:
+                model.meta.resample.pixel_scale_ratio = self.pixel_scale_ratio
+            else:
+                model.meta.resample.pixel_scale_ratio = (self.pixel_scale ** 2) / self.meta.photometry.pixelarea_arcsecsq
```

Comment:
Resetting the value of `pixel_scale_ratio` will certainly fix the situation with updating the pixel area keywords later on, but I'm wondering what might happen in the drizzle routines themselves when both `pixel_scale` and `pixel_scale_ratio` are now set to non-default values. Does the `make_output_wcs` function, for example, know enough to only pay attention to one or the other (e.g. only use `pixel_scale` when it's set)?

Reply:
yes, the presence of pscale will override pixel_scale_ratio in make_output_wcs. i will double check that this doesn't mess things up anywhere else.

Reply:
It would be helpful to have a log message reporting on HOW the value of `pixel_scale_ratio` gets set; one version reporting it was set based on the user input value of the ratio, while the other version reports that the user provided value of `pixel_scale` was used instead.   This will make it unambiguous as to what was actually used, given that a choice was made between 2 possible user inputs.

Reply:
I agree with @stsci-hack .